evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -370,12 +372,53 @@ def test_iterative_algorithm_results_best_pipeline_info_id(dummy_binary_pipeline
     scores = np.arange(0, len(next_batch))
     for pipeline_num, (score, pipeline) in enumerate(zip(scores, next_batch)):
         algo.add_result(score, pipeline, {"id": algo.pipeline_number + pipeline_num})
-    assert algo._best_pipeline_info[ModelFamily.RANDOM_FOREST]['id'] == 2
-    assert algo._best_pipeline_info[ModelFamily.LINEAR_MODEL]['id'] == 3
+    assert algo._best_pipeline_info[ModelFamily.RANDOM_FOREST]['id'] == 3
+    assert algo._best_pipeline_info[ModelFamily.LINEAR_MODEL]['id'] == 2
 
     for i in range(1, 3):
         next_batch = algo.next_batch()
         scores = -np.arange(1, len(next_batch))  # Score always gets better with each pipeline
         for pipeline_num, (score, pipeline) in enumerate(zip(scores, next_batch)):
             algo.add_result(score, pipeline, {"id": algo.pipeline_number + pipeline_num})
             assert algo._best_pipeline_info[pipeline.model_family]['id'] == algo.pipeline_number + pipeline_num
+
+
+@pytest.mark.parametrize("problem_type", [ProblemTypes.REGRESSION, ProblemTypes.BINARY, ProblemTypes.MULTICLASS])
```

Comment:
Can we add an additional test where we pass in an `_estimator_family_order` that's different from the default to ensure we use the new one rather than the default?

Reply:
We sure can! Good idea.

Given that the default works, I am gonna merge this now so that it gets into the release, and will address this comment in a separate PR.